http_logs/challenges/default.json

```diff
@@ -312,4 +328,65 @@
           "clients": 1
         }
       ]
+    },
+    {
+      "name": "append-no-conflicts-index-reindex-only",
+      "description": "Indexes the whole document corpus using Elasticsearch default settings. We only adjust the number of replicas as we benchmark a single node cluster and Rally will only start the benchmark if the cluster turns green. Document ids are unique so all index operations are append only. After indexing, same data are reindexed",
+      "schedule": [
+        {
+          "operation": "delete-index"
+        },
+        {
+          "operation": {
+            "operation-type": "create-index",
+            "settings": {{index_settings | default({}) | tojson}}
+          }
+        },
+        {
+          "name": "check-cluster-health",
+          "operation": {
+            "operation-type": "cluster-health",
+            "index": "logs-*",
+            "request-params": {
+              "wait_for_status": "{{cluster_health | default('green')}}",
+              "wait_for_no_relocating_shards": "true"
+            }
+          }
+        },
+        {
+          "operation": "index-append",
+          "warmup-time-period": 240,
+          "clients": {{bulk_indexing_clients | default(8)}}
+        },
+        {
+          "name": "refresh-after-index",
+          "operation": "refresh",
+          "clients": 1
+        },
+        {
+          "operation": "force-merge",
+          "clients": 1
+        },
+        {
+          "name": "refresh-after-force-merge",
+          "operation": "refresh",
+          "clients": 1
+        },
+        {
+          "name": "reindex",
```

Comment:
If you don't want to use a custom runner, you use the [raw-request operation](https://esrally.readthedocs.io/en/stable/track.html#raw-request).

It would then look like:

```
{
  "name": "reindex",
  "operation": {
    "operation-type": "raw-request",
    "method": "POST",
    "path": "/_reindex",
    "body": {
      "source": {
        "index": "logs-*"
      },
      "dest": {
        "index": "reindexed-logs"
      }
    },
    "request-params": {
      "request_timeout": 7200
    }
  },
  "clients": 1
}
```

Reply:
The primary purpose of the custom runner is to ensure that the `request_timeout` is passed to the python "driver" since it uses a low timeout by default. This was based on elastic/rally#567 and elastic/rally#669. I think that if I add it into request-params, it will simply be sent to the elasticsearch server as a http request-parameter?

Reply:
I see this can be confusing.

For `raw-request`, any `request_params` specified in the operation get [passed as params](https://github.com/elastic/rally/blob/89b9e791380201b6e9c855c78f7420b20b878a9b/esrally/driver/runner.py#L1250) -- by Rally -- to the [es.transport.perform_request](https://elasticsearch-py.readthedocs.io/en/master/connection.html?highlight=transport%20connection#elasticsearch.Transport.perform_request) method. In turn this method when it [detects the request_timeout key](https://github.com/elastic/elasticsearch-py/blob/35afe32a6dc34bf8cdd946a3723b59b2d0b375a4/elasticsearch/transport.py#L343) in the `params` it passes it as [timeout](https://github.com/elastic/elasticsearch-py/blob/35afe32a6dc34bf8cdd946a3723b59b2d0b375a4/elasticsearch/transport.py#L358) in the connection method.

So to summarize either for specific Rally operations that directly support setting `request_timeout` as added in elastic/rally#567 / elastic/rally#669 / your custom runner or defining it as a key in the `request-params` dict of the `raw-request` operation, ultimately it will be converted to a connection `timeout` by the Elasticsearch Python client.